noxfile.py

```diff
@@ -92,10 +125,41 @@ def default(session, install_grpc=True):
         "pytest-xdist",
     )
 
-    if install_grpc:
-        session.install("-e", ".[grpc]", "-c", constraints_path)
+    constraints_dir = str(CURRENT_DIRECTORY / "testing")
+
+    if prerelease:
+        install_prerelease_dependencies(
+            session, f"{constraints_dir}/constraints-{PYTHON_VERSIONS[0]}.txt"
+        )
+        # This *must* be the last install command to get the package from source.
+        session.install("-e", ".", "--no-deps")
     else:
-        session.install("-e", ".", "-c", constraints_path)
+        if install_grpc:
+            session.install(
+                "-e",
+                ".[grpc]",
+                "-c",
+                f"{constraints_dir}/constraints-{session.python}.txt",
+            )
+        else:
+            session.install(
+                "-e", ".", "-c", f"{constraints_dir}/constraints-{session.python}.txt"
+            )
```

Comment:
This is more succinct and clearer in highlighting the difference between the two branches, IMO:

```suggestion
        session.install(
            "-e",
            ".[grpc]" if install_grpc else ".",
            "-c",
            f"{constraints_dir}/constraints-{session.python}.txt",
            )
```